lib/spack/spack/variant.py

```diff
@@ -181,6 +181,17 @@ def variant_cls(self):
             return BoolValuedVariant
         return SingleValuedVariant
 
+    def __eq__(self, other):
+        return (self.name == other.name and
+                self.default == other.default and
+                self.values == other.values and
+                self.multi == other.multi and
+                self.single_value_validator == other.single_value_validator and
+                self.group_validator, other.group_validator)
```

Comment:
I think there's a bug here:
```suggestion
                self.group_validator == other.group_validator)
```
Unfortunately this will make a lot of other tests fail due to the check mentioned [in this comment]( https://github.com/spack/spack/pull/24858/files#r741370209)

Reply:
Solved in ca5eb36